packages/shorebird_cli/lib/src/artifact_builder.dart

```diff
@@ -256,6 +257,98 @@ class ArtifactBuilder {
     });
   }
 
+  /// Builds a macOS app using `flutter build macos`. Runs `flutter pub get`
+  /// with the system installation of Flutter to reset
+  /// `.dart_tool/package_config.json` after the build completes or fails.
+  Future<IpaBuildResult> buildMacos({
+    bool codesign = true,
+    String? flavor,
+    String? target,
+    List<String> args = const [],
+    String? base64PublicKey,
+    DetailProgress? buildProgress,
+  }) async {
+    final projectRoot = shorebirdEnv.getShorebirdProjectRoot()!;
+    // Delete the .dart_tool directory to ensure that the app is rebuilt.
+    // Without this, the build command will not print the app.dill path
+    final dartToolDir = Directory(p.join(projectRoot.path, '.dart_tool'));
+    if (dartToolDir.existsSync()) {
+      dartToolDir.deleteSync(recursive: true);
+    }
```

Comment:
Amusing.  This also "needlessly" slows down our build since it causes the source->kernel generation to have to run again.

Reply:
This could definitely be improved. A couple alternatives:

1. We could be more surgical in what we remove. I tried just deleting the app.dill file and the build failed with an error (don't have it handy)
2. We could return to our hackier way of detecting app.dill files (i.e., combing through the .dart_tool directory for the newest app.dill)

Reply:
I suspect the real fix is we'll just add this to the pile of things we fix when we move more of `shorebird` down into `flutter`